Move ListContainersForNode into cluster package

vdemeester opened the pull request:
This is purely a design refactoring so feel free to close or discuss why you think it shouldn't be done that way 👼.

It makes little sense to me to have swarm related code into the daemon package. This refactor the `daemon` and `cluster` package to remove `ListContainersForNode` from the daemon.

/cc @vieux @stevvooe @aaronlehmann 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

stevvooe commented:
LGTM, in general.

Really, though, the agent should remove the containers when leaving. This should really be handled at the swarmkit level.

vdemeester committed:
Move ListContainersForNode into cluster package

It makes little sense to have swarm related code into the daemon
package. This refactor the `daemon` and `cluster` package to remove
`ListContainersForNode` from the daemon.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

cpuguy83 commented:
LGTM
